src/contracts/minter/facets/ActionFacet.sol

```diff
@@ -158,57 +159,49 @@ contract ActionFacet is DiamondModifiers, MinterModifiers, IAction {
      * @notice Burns existing Kresko assets.
      * @param _account The address to burn kresko assets for
      * @param _kreskoAsset The address of the Kresko asset.
-     * @param _amount The amount of the Kresko asset to be burned.
+     * @param _burnAmount The amount of the Kresko asset to be burned.
      * @param _mintedKreskoAssetIndex The index of the collateral asset in the user's minted assets array.
      * @notice Only needed if withdrawing the entire deposit of a particular collateral asset.
      */
     function burnKreskoAsset(
         address _account,
         address _kreskoAsset,
-        uint256 _amount,
+        uint256 _burnAmount,
         uint256 _mintedKreskoAssetIndex
     )
         external
         nonReentrant
         kreskoAssetExistsMaybeNotMintable(_kreskoAsset)
         onlyRoleIf(_account != msg.sender, Role.MANAGER)
     {
-        if (ms().safetyStateSet) {
-            ensureNotPaused(_kreskoAsset, Action.Repay);
-        }
-        require(_amount > 0, Error.ZERO_BURN);
+        require(_burnAmount > 0, Error.ZERO_BURN);
         MinterState storage s = ms();
 
-        // Ensure the amount being burned is not greater than the user's debt.
-        uint256 debtAmount = s.kreskoAssetDebt[_account][_kreskoAsset];
-        require(_amount <= debtAmount, Error.KRASSET_BURN_AMOUNT_OVERFLOW);
-
-        {
-            // If the requested burn would put the user's debt position below the minimum
-            // debt value, close up to the minimum debt value instead.
-            FixedPoint.Unsigned memory krAssetValue = s.getKrAssetValue(_kreskoAsset, debtAmount - _amount, true);
-            if (krAssetValue.isGreaterThan(0) && krAssetValue.isLessThan(s.minimumDebtValue)) {
-                FixedPoint.Unsigned memory oraclePrice = FixedPoint.Unsigned(
-                    uint256(s.kreskoAssets[_kreskoAsset].oracle.latestAnswer())
-                );
-                FixedPoint.Unsigned memory minDebtAmount = s.minimumDebtValue.div(oraclePrice);
-                _amount = debtAmount - minDebtAmount.rawValue;
-            }
+        if (s.safetyStateSet) {
+            ensureNotPaused(_kreskoAsset, Action.Repay);
         }
 
-        // Record the burn.
-        s.kreskoAssetDebt[_account][_kreskoAsset] -= _amount;
+        uint256 debtAmount = s.getKreskoAssetDebt(_account, _kreskoAsset);
+        require(_burnAmount <= debtAmount, Error.KRASSET_BURN_AMOUNT_OVERFLOW);
+
+        // Ensure amount is either 0 or >= minDebtValue
+        uint256 burnAmountNoDust = s.ensureNotDustPosition(_kreskoAsset, _burnAmount, debtAmount);
```

Comment:
Consider throwing an error here if the `burnAmountNoDust` is 0 instead of completing the rest of the function and emitting an event with 0 assets burned.